src/client/browser/idle-page/index.js

```diff
@@ -1,5 +1,29 @@
-import IdlePage from './idle-page';
+import * as browser from '../';
+import StatusIndicator from './status-indicator';
+import COMMAND from '../../../browser-connection/command';
 
-window.init = function (statusUrl) {
-    new IdlePage(statusUrl);
+
+const CHECK_STATUS_DELAY = 1000;
+
+
+export default class IdlePage {
+    constructor (statusUrl, heartbeatUrl) {
+        this.statusIndicator = new StatusIndicator();
+        browser.startHeartbeat(heartbeatUrl, window.XMLHttpRequest);
+        this._checkStatus(statusUrl);
+    }
+
+    _checkStatus (statusUrl) {
+        browser
+            .checkStatus(statusUrl, window.XMLHttpRequest)
+            .then((cmd) => {
+                if (cmd === COMMAND.idle)
+                    window.setTimeout(() => this._checkStatus(statusUrl), CHECK_STATUS_DELAY);
+            })
+            .catch(() => this.statusIndicator.showDisconnection())
+    }
+}
+
+window.init = function (statusUrl, heartbeatUrl) {
```

Comment:
Why not expose `IdlePage` to the `window` and just call `new IdlePage(...)` in the `index.html`?